src/conn/conn_compact.c

```diff
@@ -591,13 +689,39 @@ __wt_compact_signal(WT_SESSION_IMPL *session, const char *config)
          */
         WT_ERR_MSG(
           session, EINVAL, "Background compaction is already %s", running ? "enabled" : "disabled");
-    conn->background_compact.running = !running;
 
-    /* Strip the background field from the configuration now it has been parsed. */
-    WT_ERR(__wt_config_merge(session, cfg, "background=", &stripped_config));
+    /* Update the excluded tables when the server is turned on. */
+    if (cval.val) {
+        __background_compact_clear_exclude_entry(session, false);
+        WT_ERR_NOTFOUND_OK(__wt_config_gets(session, cfg, "exclude", &cval), false);
+        if (cval.len != 0) {
+            /*
+             * Check that the configuration string only has table schema formats in the target list
+             * and construct the target hash table.
+             */
+            __wt_config_subinit(session, &exclude_config, &cval);
+            while ((ret = __wt_config_next(&exclude_config, &k, &v)) == 0) {
+                if (!WT_PREFIX_MATCH(k.str, "table:"))
+                    WT_ERR_MSG(session, EINVAL,
+                      "Background compaction can only exclude objects of type \"table\" formats in "
+                      "the exclude uri list, found %.*s instead.",
+                      (int)k.len, k.str);
+
+                WT_PREFIX_SKIP_REQUIRED(session, k.str, "table:");
+                WT_ERR(__background_compact_add_exclude_entry(
```

Comment:
How about ```__background_compact_add_exclude_entry``` be called ```__background_compact_exclude_list_add```

Reply:
Thanks! Done in cc52c025926072ed826202d782fc9641f879cd2a